src/microbe_stage/editor/PatchMapDrawer.cs

```diff
@@ -472,52 +503,166 @@ private void CreateRegionLinks()
+        if (size1 == 3)
+            priority -= 3;
+
+        // Don't connect to the sides of length 1 because connections overlap
+        if (size1 == 1 && size1 != size2)
+            priority -= 7;
+        return priority;
+    }
+
+    private int GetIShapePathCharacteristics(PatchRegion start, PatchRegion end, bool flip = false)
+    {
+        var priority = 3;
+
+        // Flip: false -> # of Columns, Flip: true -> # of Rows
+        priority = ReducePriority(flip ? start.Rows : start.Columns, flip ? start.Columns : start.Rows, priority);
+        priority = ReducePriority(flip ? end.Rows : end.Columns, flip ? end.Columns : end.Rows, priority);
+        return priority;
+    }
+
+    private (Vector2 Intermediate, int Priority) GetLShapePathCharacteristics(PatchRegion start, PatchRegion end)
+    {
+        var intermediate = new Vector2(RegionCenter(start).X, RegionCenter(end).Y);
+        var priority = 2;
+        priority = ReducePriority(start.Columns, start.Rows, priority);
+        priority = ReducePriority(end.Rows, end.Columns, priority);
+        return (intermediate, priority);
+    }
+
+    private (Vector2 Intermediate1, Vector2 Intermediate2, int Priority) GetZShapePathCharacteristics(PatchRegion start,
+        PatchRegion end, Vector2 startCenter, Vector2 endCenter, Vector2 middlePoint, bool flip = false)
+    {
+        Vector2 intermediate1;
+        Vector2 intermediate2;
+        int priority = 1;
+        if (!flip)
+        {
+            // Columns
+            intermediate1 = new Vector2(startCenter.X, middlePoint.Y);
+            intermediate2 = new Vector2(endCenter.X, middlePoint.Y);
+        }
+        else
+        {
+            // Rows
+            intermediate1 = new Vector2(middlePoint.X, startCenter.Y);
+            intermediate2 = new Vector2(middlePoint.X, endCenter.Y);
+        }
+
+        priority = ReducePriority(flip ? start.Rows : start.Columns, flip ? start.Columns : start.Rows, priority);
+        priority = ReducePriority(flip ? end.Rows : end.Columns, flip ? start.Columns : start.Rows, priority);
+        return (intermediate1, intermediate2, priority);
+    }
+
+    private (Vector2 Intermediate1, Vector2 Intermediate2, int Priority) GetUShapePathCharacteristics(PatchRegion start,
+        PatchRegion end, Rect2 startRect, Rect2 endRect, Vector2 startCenter, Vector2 endCenter, string shape, int i)
+    {
+        var upper = startRect.Position.Y < endRect.Position.Y ? startRect : endRect;
+        var lower = startRect.End.Y > endRect.End.Y ? startRect : endRect;
+        var left = startRect.Position.X < endRect.Position.X ? startRect : endRect;
+        var right = startRect.End.X > endRect.End.X ? startRect : endRect;
+        Vector2 intermediate1;
+        Vector2 intermediate2;
+        int priority = 0;
+        switch (shape)
+        {
+            case "lower":
+                intermediate1 = new Vector2(startCenter.X, lower.End.Y + i * 50);
```

Comment:
Case statements with multiple lines should be wrapped in braces according to our styleguide.